supplier_app/views.py

```diff
@@ -126,15 +127,35 @@ def save_company(self, forms):
 
     def get_success_url(self):
         return reverse('company-list')
-    
+
     def get_failure_url(self):
         messages.error(
-                self.request,
-                COMPANY_ALREADY_EXIST,
-            )
+            self.request,
+            COMPANY_ALREADY_EXIST,
+        )
         return reverse('company-create')
 
 
+class CompanyManage(LoginRequiredMixin, ListView):
+    model = User
+    template_name = 'supplier_app/supplier/company_manage.html'
+
+    def get_queryset(self):
+        company = CompanyUserPermission.objects.filter(user_id=self.request.user.id).first()
```

Comment:
you can use`get` instead of `filter.first`

Reply:
That cannot be possible, because this query return more than 1 result

Reply:
So there are cases where a user has more than one companyUserPermission, but how do you know that you only need the first one?

Reply:
We assume that the first company that the user registers is correct.
There is an legacy bug that allows duplicate insertion in that table, they are always the same values, that's why we always take the first one. When that bug is solved it will be safe to use get function.

Reply:
Ok so I recommend to create a ticket(if it does not exists a ticket yet) to track that bug for future references. Thanks!